src/Airship/Internal/Decision.hs

```diff
@@ -559,12 +559,17 @@ l13 r@Resource{..} = do
         Nothing ->
             m16 r
 
-l07 r = do
+l07 r@Resource{..} = do
     trace "l07"
     req <- lift request
-    if requestMethod req == HTTP.methodPost
-        then m07 r
-        else lift $ halt HTTP.status404
+    if requestMethod req == HTTP.methodPost then m07 r
+    else do
+        m <- _contentType <$> get
+        when (isJust m) $ do
+            let (Just (ct, body)) = m
+            lift body >>= lift . putResponseBody
+            lift $ addResponseHeader ("Content-Type", renderHeader ct)
```

Comment:
I agree that it does feel a little strange. Another difference from Erlang webmachine is that it lets you define responses for any error type rather than just 404. I wonder if rather than a Resource  that is only intended for 404 responses if it might be better to accept some mapping of status codes to responses. If no mapping is provided by the user for a particular error status then no response body is set.